ecgtools/parsers.py

```diff
@@ -92,3 +100,252 @@ def cmip6_default_parser(
+                    if 'glob_string' not in stream_info.keys():
+                        print("ERROR: Each ensemble must contain a 'glob_string' key.")
+                        return False
+        return True
+
+    def _validate_yaml(self):
+        """
+        Validates the generic yaml input against the schema.  It uses either yamale or the internal validater.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        boolean
+            True - passes the validation, False - fails the validation
+        """
+
+        # verify the format is correct
+        if self.validater == 'yamale':
+            try:
+                import yamale
+
+                print('Validating yaml file with yamale.')
+                cwd = Path(os.path.dirname(__file__))
+                schema_path = str(cwd.parent / 'schema') + '/generic_schema.yaml'
+                schema = yamale.make_schema(schema_path)
+                data = yamale.make_data(self.yaml_path)
+                try:
+                    yamale.validate(schema, data)
+                    print('Validation success! 👍')
+                    return True
+                except ValueError as e:
+                    print(
+                        'Yamale found that your file, '
+                        + self.yaml_path
+                        + ' is not formatted correctly.'
+                    )
+                    print(e)
+                    return False
+            except ImportError:
+                print('Validating yaml file internally.')
+                return self._internal_validation()
+        else:
+            print('Validating yaml file internally.')
+            return self._internal_validation()
+
+    def _parser_netcdf(self, filepath, local_attrs):
+        """
+        Opens a netcdf file in order to gather time and requested attribute information.
+        Also attaches assigned attributes gathered from the yaml file.
+
+        Parameters
+        ----------
+        filepath : str
+            The full path to the netcdf file to attatch attributes to.
+        local_attrs : dict
+            Holds attributes that need to be attached to the filenpath.
+
+        Returns
+        -------
+        dict
+            Returns all of the attributes that need to be assigned to the netcdf.
+        """
+
+        fileparts = {}
+        fileparts['path'] = filepath
+
+        try:
+            fileparts['variable'] = []
+            # open file
+            d = nc.Dataset(filepath, 'r')
+            # find what the time (unlimited) dimension is
+            dims = list(dict(d.dimensions).keys())
+            if 'time' in d.variables.keys():
+                times = d['time']
+                start = str(times[0])
+                end = str(times[-1])
+                fileparts['time_range'] = start + '-' + end
+            # loop through all variables
+            for v in d.variables.keys():
```

Comment:
Isn't this the same as `for v in d.variables`?

Reply:
Yes and thanks.  I went ahead and made that change as well.